src/main/java/com/couchbase/lite/replicator/ReplicationInternal.java

```diff
@@ -278,7 +278,8 @@ public void databaseClosing() {
     protected void close() {
         // shutdown ScheduledExecutorService. Without shutdown, cause thread leak
         if (remoteRequestExecutor != null && !remoteRequestExecutor.isShutdown()) {
-            Utils.shutdownAndAwaitTermination(remoteRequestExecutor);
+            // Note: Time to wait is set 60 sec because RemoteRequest's socket timeout is set 60 seconds.
+            Utils.shutdownAndAwaitTermination(remoteRequestExecutor, 60);
         }
```

Comment:
Switch from HttpClient to openConneciton() is another big change. We can discuss it on Monday.

Lollipop SDK 5.0.x is API 21. And we need to to support all versions since API 9 SDK (2.?.?). 
So HttpClient will be deprecated, but for short term, `HttpClient.getConnectionManager().shutdown()` could be solution.